src/reshuffle.jl

```diff
@@ -0,0 +1,36 @@
+
+"""
+    ReshuffleIter(iter)
+
+Wrap a data iterator `iter` created by [`eachobs`](@ref) so that it is
+shuffled anew every time it is iterated over.
+
+This is an internal function. Use `eachobs(data; shuffle = true)` instead.
+"""
+struct ReshuffleIter{T}
+    iter::T
+end
```

Comment:
Do we need this type? Since we are creating a `EachObs` we could store the `shuffle` option in there.

Reply:
See above, then I'd suggest to make a single `EachObs` type that holds these flags

Reply:
I agree, except I would like to expand this a bit. See my comment above.